api/externals/repository/division_repository.go

```diff
@@ -0,0 +1,142 @@
+package repository
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/NUTFes/FinanSu/api/drivers/db"
+	"github.com/NUTFes/FinanSu/api/externals/repository/abstract"
+	"github.com/NUTFes/FinanSu/api/generated"
+	goqu "github.com/doug-martin/goqu/v9"
+)
+
+type divisionRepository struct {
+	client db.Client
+	crud   abstract.Crud
+}
+
+type DivisionRepository interface {
+	AllByPeriodAndFinancialRecord(context.Context, string, string) (*sql.Rows, error)
+	GetById(context.Context, string) (*sql.Row, error)
+	Create(context.Context, division) error
+	Update(context.Context, string, division) error
+	Delete(context.Context, string) error
+	FindLatestRecord(context.Context) (*sql.Row, error)
+}
+
+func NewDivisionRepository(c db.Client, ac abstract.Crud) DivisionRepository {
+	return &divisionRepository{c, ac}
+}
+
+// 年度別と財務記録で取得
+func (dr *divisionRepository) AllByPeriodAndFinancialRecord(
+	c context.Context,
+	year string,
+	financialRecordId string,
+) (*sql.Rows, error) {
+
+	ds := selectDivisionQuery
+
+	if year != "" {
+		ds = ds.Where(goqu.C("years.year").Eq(year))
+	}
+
+	if financialRecordId != "" {
+		ds = ds.Where(goqu.C("financial_records.id").Eq(financialRecordId))
+	}
```

Comment:
ここsqlでsyntaxエラー起きるので修正お願いします
これだとうまくいきました
```suggestion
	if year != "" {
		ds = ds.Where(goqu.Ex{"years.year": year})
	}

	if financialRecordId != "" {
		ds = ds.Where(goqu.Ex{"financial_records.idr": financialRecordId})
	}
```